src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,84 @@
+package duke.task;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+
+import org.json.simple.JSONObject;
+
+public class Deadline extends Task {
+
+    private String by;
+
+    /**
+     * Default constructor for a Deadline.
+     *
+     * @param description The description of the Deadline.
+     * @param by The deadline of the Deadline task.
+     */
+    protected Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * A constructor for a Deadline which includes a boolean isCompleted.
+     *
+     * @param description The description of the Deadline.
+     * @param by The deadline of the Deadline task.
+     * @param isCompleted Whether the Deadline has been completed.
+     */
+    protected Deadline(String description, String by, boolean isCompleted) {
+        super(description);
+        this.by = by;
+        this.isCompleted = isCompleted;
+    }
+
+    /**
+     * Default factory method for a Deadline.
+     *
+     * @param description The description of the Deadline.
+     * @param by The deadline of the Deadline task.
+     * @return A new Deadline with the given description.
+     */
+    public static Deadline of(String description, String by) {
+        return new Deadline(description, by);
+    }
+
+    /**
+     * A factory method for a Deadline which includes a boolean isCompleted.
+     *
+     * @param description The description of the Deadline.
+     * @param by The deadline of the Deadline task.
+     * @param isCompleted Whether the Deadline has been completed.
+     * @return A new Deadline with the given description.
+     */
+    public static Deadline of(String description, String by, boolean isCompleted) {
+        return new Deadline(description, by, isCompleted);
+    }
+
+
+    @Override
+    public String toString() throws DateTimeException {
```

Comment:
By the looks of it, if the user inputs an invalid date, this will cause this error to be thrown causing the program to stop. Perhaps catch the error and return something more useful to the user such as how to format the date properly to make the program work.